revocations.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 The HIBA Authors
+ *
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+#ifndef _REVOCATIONS_H
+#define _REVOCATIONS_H
+
+#include <sys/types.h>
+
+#include "sshbuf.h"
+
+/*
+ * HIBA GRL magic header
+ */
+#define HIBA_GRL_MAGIC 0x4847524c
+
+struct hibagrl;
+
+/* Allocate a hibagrl object.
+ * Result must be freed using the SSH API hibagrl_free. */
+struct hibagrl *hibagrl_new();
+
+/* Initializes an HIBA GRL object.
+ * This hibagrl object is mapped and ready to accept new revocations. */
+void hibagrl_init(struct hibagrl *grl, const char *comment);
+
+/* Load an existing serialzied grant revocation list.
+ * This hibagrl is flat and ready to accept checks.
+ * The sshbuf must not be modified or freed while the hibagrl object is used. */
+int hibagrl_decode(struct hibagrl *grl, struct sshbuf *blob);
+
+/* Releases resources allocated by a hibagrl object.
+ * This function supports both flat and mapped objects. */
+void hibagrl_free();
+
+/* Take a flat hibagrl object and map it for editing. */
+int hibagrl_map(struct hibagrl *grl);
+
+/* Serializes a flat grant revocation list.*/
+int hibagrl_encode(const struct hibagrl *grl, struct sshbuf *blob);
```

Comment:
This is actually only for mapped, GRL, not flat ones.
I updated the commment.

FWIW, serialized a flat GRL is ot making much sense since it is read only, so one might as well just write back the buffer used initially.